### DISCUSSION TITLE
edit-in-kitty source ~/.zshrc first?

### DISCUSSION THREAD

**Discussion by sinkr** [kovidgoyal/kitty#7417] [Q&A] [answered]
Love Kitty, just getting started, but I am noticing that when I use the `edit-in-shell` kitten on a remote machine, neovim comes up somewhat uninitialized.

For instance, my .zshrc pulls my Copilot API token out of the macOS keychain, which isn't present.

I have 2 questions, actually:

1. Is it possible to source .zshrc prior to opening nvim, and/or
2. is it possible to send the file to an already open copy of NeoVim?

Thanks again for the great terminal!

**Comment by kovidgoyal** [comment]
What's the edit-in-shell kitten? If you want to edit a remote file
in your local nvim setup, you use the ssh kitten to ssh into the remote
computer and then run the command

edit-in-kitty filename

See
https://sw.kovidgoyal.net/kitty/shell-integration/#edit-files-in-new-kitty-windows-even-over-ssh

**Reply by sinkr** [reply]
Sorry, I meant edit-in-kitty.  I’ve been through the docs and can’t quite make it work right macOS arm64 Sonoma -> macOS arm64 Sonoma or to a Linux box.

**Reply by kovidgoyal** [reply]
edit-in-kitty will simply run whatever you have editor set to. If you need it to do something else before running your actual editor, simply change the value of editor in kitty.conf to be a shell script that does whatever you need before execing the actual editor.

**Reply by sinkr** [reply]
Thank you for the responses, but I can't make it work correctly after hours of hacking away at it.  macOS -> Linux, I get an X11 error for some reason (attached below) and macOS -> macOS, I get Bourne shell errors (sh, not bash) and I have my shell and editor explicitly set in kitty.conf:
```
shell /opt/homebrew/bin/zsh --interactive --login
editor /usr/local/bin/nvim
```

macOS -> Linux error:
```
[129 02:23:31.286816] Ignoring unknown config key: show_hyperlink_targets
[129 02:23:31.286837] Ignoring unknown config key: underline_hyperlinks
[129 02:23:31.287347] Ignoring invalid config line: file_transfer_confirmation_bypass
[129 02:23:31.287367] Ignoring unknown config key: shell_integration
[129 02:23:31.287379] Ignoring unknown config key: allow_cloning
[129 02:23:31.289689] [glfw error 65544]: X11: The DISPLAY environment variable is missing
GLFW initialization failed                                                                
```

**Comment by kovidgoyal** [comment]
I have no idea what you are doing, you appear to somehow be trying to
run kitty on your Linux machine.

I suggest you make sure kitty is either not installed or is up to date
on your Linux machine, that means using the kitty official binary
install not whatever kitty package the linux distro provides.

Then in kitty on your mac machine do:

kitten ssh my-linux-machine
my-linux machine $ edit-in-kitty file.txt

this will open file.txt on your *mac* machine in nvim also running on
your mac machine.

Note that the first time you run edit-in-kitty assuming kitty is not
installed on your Linux machine, it will download the kitten binary.

Good luck.

**Reply by sinkr** [reply]
OK, so I hosed it out and it did as you said and installed, but it still is trying to do X11.  I'm giving up at this point.  I hoped to have a way to do this easily, but it doesn't seem possible.  Back to netrw.
```
edit-in-kitty testfile                                                              
kitty needs to be installed

Downloading kitty from: https://github.com/kovidgoyal/kitty/releases/download/v0.34.1/kitty-0.34.1-x86_64.txz

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 21.0M  100 21.0M    0     0  14.5M      0  0:00:01  0:00:01 --:--:-- 17.6M
[0.039] [glfw error 65544]: X11: The DISPLAY environment variable is missing
GLFW initialization failed                                                         
```

**Comment by kovidgoyal** [comment]
That implies your edit-in-kitty alias is incorrect.

It should be: kitten edit-in-kitty

Either fix it or explicitly run

kitten edit-in-kitty

instead.

**Reply by sinkr** [reply]
I appreciate your time.

I hosed out the directory again, reinstalled and directly referenced the newly-installed binary to no avail:
```
curl -L https://sw.kovidgoyal.net/kitty/installer.sh | sh /dev/stdin
.local/share/kitty-ssh-kitten/kitty/bin/kitty  edit-in-kitty testfile                      
[0.040] [glfw error 65544]: X11: The DISPLAY environment variable is missing
GLFW initialization failed
```

**Comment by kovidgoyal** [comment]
No, dont run

kitty edit-in-kitty file

run just

edit-in-kitty file

**Comment by sinkr** [comment]
I get the same thing:
```
edit-in-kitty testfile                                                                    
[0.039] [glfw error 65544]: X11: The DISPLAY environment variable is missing
GLFW initialization failed         
```

**Comment by kovidgoyal** [comment]
that's impossible unless as I said before your edit-in-kitty alias is wrong.

So run

kitten edit-in-kitty file

And note this is KITTEN not KITTY

**Comment by sinkr** [comment, marked as answer]
> that's impossible unless as I said before your edit-in-kitty alias is wrong. So run kitten edit-in-kitty file And note this is KITTEN not KITTY

Holy mackerel that was it!  It was so subtle I kept overlooking it.

On macOS, the aliases were automatically set up for me, but on Linux I had to do them myself and apparently muffed it.

@kovidgoyal 's assertion above is **real** the solution, but I want to provide full context for anyone else having this or similar issues.

My problem was 2 things:

1. The alias on my Ubuntu Linux machine was incorrect.  Both `apt` and the `curl` install methods do not normalize the shell for you automatically like what seemingly occurs on macOS via `brew`.
2. I did not have my `editor` overridden in `kitty.conf`; it was using an older version of NeoVim and not initializing my `.zshrc`.
